run_neuron.py

```diff
@@ -0,0 +1,82 @@
+
+        if auto_update:
+            current_branch = subprocess.getoutput("git rev-parse --abbrev-ref HEAD")
+            local_commit = subprocess.getoutput("git rev-parse HEAD")
+            os.system("git fetch")
+            remote_commit = subprocess.getoutput(f"git rev-parse origin/{current_branch}")
+
+            if should_update_local(local_commit, remote_commit):
+                print("Local repo is not up-to-date. Updating...")
+                reset_cmd = "git reset --hard " + remote_commit
+                process = subprocess.Popen(reset_cmd.split(), stdout=subprocess.PIPE)
+                output, error = process.communicate()
+
+                if error:
+                    print("Error in updating:", error)
+                else:
+                    print("Updated local repo to latest version: {}", format(remote_commit))
+                    
+                    print("Running the autoupdate steps...")
+                    # Trigger shell script. Make sure this file path starts from root
+                    os.system(f"./autoupdate_{neuron_type}_steps.sh")
+                    time.sleep(20)
+                    print("Finished running the autoupdate steps 😎")
+                    print("Restarting neuron")
+                    os.system(f"./start_{network}_{neuron_type}.sh")
+            else:
+                print("Repo is up-to-date.")
+
+        if self_heal:
+            # Check if it's time to restart the PM2 process
+            if time.time() - last_restart_time >= RESTART_INTERVAL_HOURS * 3600:
+                os.system(f"./start_{network}_{neuron_type}.sh")
+                last_restart_time = time.time()  # Reset the timer after the restart
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(description="Bittensor neuron run script with optional self-healing and auto-update.")
+    parser.add_argument("--validator", action="store_true")
+    parser.add_argument("--miner", action="store_true")
+    parser.add_argument("--network", default="finney", choices=["finney", "test"], help="finney or test")
+    parser.add_argument("--no-self-heal", action="store_true", help="Disable the automatic restart of the PM2 process")
+    parser.add_argument("--no-auto-update", action="store_true", help="Disable the automatic update of the local repository")
+
+    args = parser.parse_args()
+    if not (args.miner ^ args.validator):
+        print(f"Usage: python {__file__}" +  "--validator | --miner [--network {finney,test} --no-self-heal --no-auto-update]")
+        sys.exit(1)
+
+    neuron_type = 'miner' if args.miner else 'validator'
+    network = 'mainnet' if args.network =='finney' else 'testnet'
+
+    os.system(f"./start_{network}_{neuron_type}.sh")
+
+    if not args.no_auto_update or not args.no_self_heal:
+        run_auto_update_self_heal(
+            neuron_type,
+            network,
+            auto_update=not args.no_auto_update,
+            self_heal=not args.no_self_heal)
```

Comment:
What was the reasoning behind this conditional logic? Is this a common pattern used in other subnets?

Reply:
It does look a bit funny - I wanted to have the auto updates and self healing be the default behavior, and this is the conditional logic that follows.

Reply:
i could make the args to `run_auto_update_self_heal` `no_auto_update` and `no_self_heal` instead, that'd be cleaner on lines 80 and 81